File: src/main/java/vip/yazilim/libs/springcore/exception/general/database/DatabaseDeleteException.java
package vip.yazilim.libs.springcore.exception.general.database;

import java.util.Arrays;

public class DatabaseDeleteException extends DatabaseException {
	private static final long serialVersionUID = 7558357250959860512L;

	public <E, ID> DatabaseDeleteException(Class<E> entityClass, ID id, Exception exception) {
		super(entityClass, "DELETE", String.valueOf(id), exception);
	}

	public <E, ID> DatabaseDeleteException(Class<E> entityClass, Exception exception, Object... params) {
		super(entityClass, "DELETE", Arrays.toString(params), exception);
	}
}

File: src/main/kotlin/vip/yazilim/libs/springvip/util/generic/service/exception/DatabaseDeleteException.kt
package vip.yazilim.libs.springvip.util.generic.service.exception

import kotlin.reflect.KClass

// DELETE
class DatabaseDeleteException : DatabaseException {
    constructor(entityClass: KClass<*>, id: Any, exception: Exception)
            : super(entityClass, "DELETE", id.toString(), exception)

    constructor(entityClass: KClass<*>, exception: Exception, vararg params: Any?)
            : super(entityClass, "DELETE", params.contentToString(), exception)
}